keel-core/src/main/kotlin/com/netflix/spinnaker/keel/extensions/DefaultExtensionRegistry.kt

```diff
@@ -0,0 +1,34 @@
+package com.netflix.spinnaker.keel.extensions
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.fasterxml.jackson.databind.jsontype.NamedType
+import com.netflix.spinnaker.keel.api.support.ExtensionRegistry
+import org.slf4j.LoggerFactory
+import org.springframework.stereotype.Component
+
+@Component
+class DefaultExtensionRegistry(
+  private val mappers: List<ObjectMapper>
+) : ExtensionRegistry {
+  private val baseToExtensionTypes = mutableMapOf<Class<*>, MutableMap<String, Class<*>>>()
+
+  override fun <BASE : Any> register(
+    baseType: Class<BASE>,
+    extensionType: Class<out BASE>,
+    discriminator: String
+  ) {
+    baseToExtensionTypes
+      .getOrPut(baseType, ::mutableMapOf)
+      .also { it[discriminator] = extensionType }
+    log.info("Registering extension \"$discriminator\" for ${baseType.simpleName} using ${extensionType.simpleName}")
+    mappers.forEach {
+      it.registerSubtypes(NamedType(extensionType, discriminator))
+    }
+  }
```

Comment:
Default implementation does the Jackson registration right away and stores the subtype mapping for later use.